### PR TITLE
Parser returns input output tables

### PR DESCRIPTION
**Pull request by typhoonzero**
Part work of https://github.com/sql-machine-learning/sqlflow/issues/2016

### CODE REVIEW

**Comment by Yancey0623** [comment]
#2000 is a design doc, how about create a TODO issue?

File: java/parser/src/main/proto/Parser.proto [hunk 1]
```diff
@@ -25,6 +25,12 @@ message ParserRequest {
   string sql_program = 2;
 }
 
+// InputOutputTables represents input and output tables a statement manipulates.
+message InputOutputTables {
+  repeated string input_tables = 1;
```
Comment (weiguoz) [inline java/parser/src/main/proto/Parser.proto]: `input_tables` => `inputs`?

File: java/parser/src/main/proto/Parser.proto [hunk 2]
```diff
@@ -25,6 +25,12 @@ message ParserRequest {
   string sql_program = 2;
 }
 
+// InputOutputTables represents input and output tables a statement manipulates.
+message InputOutputTables {
```
Comment (weiguoz) [inline java/parser/src/main/proto/Parser.proto:29]: I slightly suggest shortening the name to `IOTables`, like `<iostream>` in CPP

File: java/parse-interface/src/main/java/org/sqlflow/parser/parse/InputOutputTables.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.sqlflow.parser.parse;
+
+import java.util.ArrayList;
+
+public class InputOutputTables {
+  public ArrayList<String> inputTables;
```
Comment (weiguoz) [inline java/parse-interface/src/main/java/org/sqlflow/parser/parse/InputOutputTables.java]: ArrayList -> List ?

Comment (typhoonzero) [reply java/parse-interface/src/main/java/org/sqlflow/parser/parse/InputOutputTables.java]: Thanks, I also checked out this post: https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java

File: java/parser/src/main/proto/Parser.proto (hunk 2 above)
Comment (typhoonzero) [reply java/parser/src/main/proto/Parser.proto:29]: The name `IOTables` causes:

```
[WARN] /Users/wuyi/go/src/sqlflow.org/sqlflow/java/parse-interface/src/main/java/org/sqlflow/parser/parse/ParseResult.java:15:30: Member name 'IOTables' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]
[WARN] /Users/wuyi/go/src/sqlflow.org/sqlflow/java/parse-interface/src/main/java/org/sqlflow/parser/parse/ParseResult.java:15:30: 名称 'IOTables' 中不能出现超过 '2' 个连续大写字母。 [AbbreviationAsWordInName]
```

File: java/parse-interface/src/main/java/org/sqlflow/parser/parse/ParseResult.java [hunk 4]
```diff
@@ -11,4 +11,6 @@
   public int position;
   // Errors encountered during parsing.
   public String error;
+  // tables that each statement manipulates.
+  public ArrayList<InputOutputTables> inputOutputTables;
```
Comment (weiguoz) [inline java/parse-interface/src/main/java/org/sqlflow/parser/parse/ParseResult.java]: Similarly, use `List`

Comment (typhoonzero) [reply java/parse-interface/src/main/java/org/sqlflow/parser/parse/ParseResult.java]: Original `public ArrayList<String> statements;` is still using `ArrayList` will modify this in another PR.

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**